inference-engine/src/mkldnn_plugin/nodes/mkldnn_priorbox_clustered_node.cpp

```diff
@@ -0,0 +1,178 @@
+    const ngraph::opset1::PriorBoxClustered::Attributes& attrs = priorBox->get_attrs();
+
+    widths = attrs.widths;
+    heights = attrs.heights;
+    clip = attrs.clip;
+    variances = attrs.variances;
+    step = attrs.step;
+    step_heights = attrs.step_heights;
+    step_widths = attrs.step_widths;
+    offset = attrs.offset;
+
+    number_of_priors = widths.size();
+
+    if (variances.empty()) {
+        variances.push_back(0.1f);
+    }
+}
+
+bool MKLDNNPriorBoxClusteredNode::needShapeInfer() const {
+    auto& memory = getChildEdgeAt(0)->getMemoryPtr();
+    if (memory->GetShape().isDynamic()) {
+        return true;
+    }
+
+    const auto& outputShape = memory->GetShape().getStaticDims();
+    const int* in_data = reinterpret_cast<int*>(memory->GetPtr());
+    const int h = in_data[0];
+    const int w = in_data[1];
+    const auto output = static_cast<size_t>(4 * h * w * number_of_priors);
+
+    return outputShape[1] != output;
+}
+
+std::vector<VectorDims> MKLDNNPriorBoxClusteredNode::shapeInfer() const {
+    const int* in_data = reinterpret_cast<int*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr());
+    const int H = in_data[0];
+    const int W = in_data[1];
+    const auto output = static_cast<size_t>(4 * H * W * number_of_priors);
+    return {{2, output}};
+}
+
+bool MKLDNNPriorBoxClusteredNode::needPrepareParams() const {
+    return false;
+}
+
+void MKLDNNPriorBoxClusteredNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    addSupportedPrimDesc(
+            {{LayoutType::ncsp, Precision::I32}, {LayoutType::ncsp, Precision::I32}},
+            {{LayoutType::ncsp, Precision::FP32}},
+            impl_desc_type::ref_any);
+}
+
+void MKLDNNPriorBoxClusteredNode::createPrimitive() {
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
+    }
+}
+
+void MKLDNNPriorBoxClusteredNode::execute(mkldnn::stream strm) {
+    const int* in_data = reinterpret_cast<int*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr());
+    const int layer_height = in_data[0];
+    const int layer_width = in_data[1];
+
+    const int* in_image = reinterpret_cast<int*>(getParentEdgeAt(1)->getMemoryPtr()->GetPtr());
+    int img_height = in_image[0];
+    int img_width = in_image[1];
+
+    float step_w = step_widths == 0 ? step : step_widths;
+    float step_h = step_heights == 0 ? step : step_heights;
+    if (step_w == 0 && step_h == 0) {
+        step_w = static_cast<float>(img_width) / layer_width;
+        step_h = static_cast<float>(img_height) / layer_height;
+    }
+
+    float* dst_data = reinterpret_cast<float*>(getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+    const auto& out_shape = getChildEdgeAt(0)->getMemory().GetShape().getStaticDims();
+
+    size_t var_size = variances.size();
+    parallel_for(layer_height, [&](int64_t h) {
```

Comment:
is it possible to use `parallel_for2d` here and parallel by `h` and `w` ?

Reply:
done, thanks